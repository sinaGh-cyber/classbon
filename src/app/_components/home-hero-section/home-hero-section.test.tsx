import { HomeHeroSection } from "./home-hero-section";
import { render } from "@testing-library/react";

describe("Home hero section", () => {
    test("display the home hero section", () => {
        const { container } = render(<HomeHeroSection />);
        expect(container)
          .toMatchSnapshot(`<section className="bg-hero-pattern bg-no-repeat bg-center xl:bg-left mt-5 xl:mt-20">
        <div className="container flex flex-col-reverse items-center xl:flex-row">
          <div className="flex-1 flex flex-col gap-5 mt-12 pb-5 text-center xl:text-right">
            <h3 className="text-xl xl:text-2xl dark:text-info">
              خوش اومدی به ...
            </h3>
            <h1 className=" text-3xl lg:text-5xl xl:text-5xl font-black gradient">
              مسیر صعود به قله‌های برنامه‌نویسی
            </h1>
            <p className="max-w-2xl  text-lg md:text-xl font-bold leading-8">
              هر جای مسیرِ برنامه‌نویسی که باشی، با هم‌راهی استادهای باتجربهٔ
              کلاسبن می‌تونی بدون محدودیت به قله‌های بالاتر صعود کنی. ما همیشه
              هواتو داریم.
            </p>
            <div className="mt-5 flex gap-4 justify-center xl:justify-start">
              <Button variant="primary" size="large">
                دوره‌های ری‌اکت و نکست
                <IconArrowLeftFill fill="currentColor" />
              </Button>
              <Button variant="neutral" size="large">
                مشاوره برنامه‌نویسی
              </Button>
            </div>
            <Image
              src="/images/frameworks.png"
              className="grayscale mt-4 opacity-70 m-auto xl:m-0"
              width={412}
              height={39}
              alt=""
            />
          </div>
          <Image
            src="/images/programmer-landing.svg"
            alt=""
            width={702}
            height={521}
          />
        </div>
      </section>`);
    });
});