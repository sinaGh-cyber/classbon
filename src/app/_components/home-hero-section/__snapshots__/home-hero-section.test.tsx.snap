// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home hero section display the home hero section: 
        <section className="bg-hero-pattern bg-no-repeat bg-center xl:bg-left mt-5 xl:mt-20">
            <div className="container flex flex-col-reverse items-center xl:flex-row">
                <div className="flex-1 flex flex-col gap-5 mt-12 pb-5 text-center xl:text-right">
                    <h3 className="text-xl xl:text-2xl dark:text-info">
                        خوش اومدی به ...
                    </h3>
                    <h1 className=" text-3xl lg:text-5xl xl:text-5xl font-black gradient">
                        مسیر صعود به قله‌های برنامه‌نویسی
                    </h1>
                    <p className="max-w-2xl  text-lg md:text-xl font-bold leading-8">
                        هر جای مسیرِ برنامه‌نویسی که باشی، با هم‌راهی استادهای
                        باتجربهٔ کلاسبن می‌تونی بدون محدودیت به قله‌های بالاتر
                        صعود کنی. ما همیشه هواتو داریم.
                    </p>
                    <div className="mt-5 flex gap-4">
                        <Button variant="primary" size="large">
                            دوره‌های ری‌اکت و نکست
                        </Button>
                        <Button variant="neutral" size="large">
                            مشاوره برنامه‌نویسی
                        </Button>
                    </div>
                    <Image
                        src="/images/frameworks.png"
                        className="grayscale mt-4 opacity-70 m-auto xl:m-0"
                        width={412}
                        height={39}
                        alt=""
                    />
                </div>
                <Image
                    src="/images/programmer-landing.svg"
                    alt=""
                    width={702}
                    height={521}
                />
            </div>
        </section>        
         1`] = `
<div>
  <section
    class="bg-hero-pattern bg-no-repeat bg-center xl:bg-left mt-5 xl:mt-20"
  >
    <div
      class="container flex flex-col-reverse items-center xl:flex-row"
    >
      <div
        class="flex-1 flex flex-col gap-5 mt-12 pb-5 text-center xl:text-right"
      >
        <h3
          class="text-xl xl:text-2xl dark:text-info"
        >
          خوش اومدی به ...
        </h3>
        <h1
          class=" text-3xl lg:text-5xl xl:text-5xl font-black gradient"
        >
          مسیر صعود به قله‌های برنامه‌نویسی
        </h1>
        <p
          class="max-w-2xl  text-lg md:text-xl font-bold leading-8"
        >
          هر جای مسیرِ برنامه‌نویسی که باشی، با هم‌راهی استادهای باتجربهٔ کلاسبن می‌تونی بدون محدودیت به قله‌های بالاتر صعود کنی. ما همیشه هواتو داریم.
        </p>
        <div
          class="mt-5 flex gap-4"
        >
          <button
            class="btn btn-primary btn-lg"
            type="button"
          >
            دوره‌های ری‌اکت و نکست
          </button>
          <button
            class="btn btn-neutral btn-lg"
            type="button"
          >
            مشاوره برنامه‌نویسی
          </button>
        </div>
        <img
          alt=""
          class="grayscale mt-4 opacity-70 m-auto xl:m-0"
          data-nimg="1"
          decoding="async"
          height="39"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75"
          srcset="/_next/image?url=%2Fimages%2Fframeworks.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75 2x"
          style="color: transparent;"
          width="412"
        />
      </div>
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="521"
        loading="lazy"
        src="/images/programmer-landing.svg"
        style="color: transparent;"
        width="702"
      />
    </div>
  </section>
</div>
`;

exports[`Home hero section display the home hero section: <section className="bg-hero-pattern bg-no-repeat bg-center xl:bg-left mt-5 xl:mt-20">
        <div className="container flex flex-col-reverse items-center xl:flex-row">
          <div className="flex-1 flex flex-col gap-5 mt-12 pb-5 text-center xl:text-right">
            <h3 className="text-xl xl:text-2xl dark:text-info">
              خوش اومدی به ...
            </h3>
            <h1 className=" text-3xl lg:text-5xl xl:text-5xl font-black gradient">
              مسیر صعود به قله‌های برنامه‌نویسی
            </h1>
            <p className="max-w-2xl  text-lg md:text-xl font-bold leading-8">
              هر جای مسیرِ برنامه‌نویسی که باشی، با هم‌راهی استادهای باتجربهٔ
              کلاسبن می‌تونی بدون محدودیت به قله‌های بالاتر صعود کنی. ما همیشه
              هواتو داریم.
            </p>
            <div className="mt-5 flex gap-4">
              <Button variant="primary" size="large">
                دوره‌های ری‌اکت و نکست
              </Button>
              <Button variant="neutral" size="large">
                مشاوره برنامه‌نویسی
              </Button>
            </div>
            <Image
              src="/images/frameworks.png"
              className="grayscale mt-4 opacity-70 m-auto xl:m-0"
              width={412}
              height={39}
              alt=""
            />
          </div>
          <Image
            src="/images/programmer-landing.svg"
            alt=""
            width={702}
            height={521}
          />
        </div>
      </section> 1`] = `
<div>
  <section
    class="bg-hero-pattern bg-no-repeat bg-center xl:bg-left mt-5 xl:mt-20"
  >
    <div
      class="container flex flex-col-reverse items-center xl:flex-row"
    >
      <div
        class="flex-1 flex flex-col gap-5 mt-12 pb-5 text-center xl:text-right"
      >
        <h3
          class="text-xl xl:text-2xl dark:text-info"
        >
          خوش اومدی به ...
        </h3>
        <h1
          class=" text-3xl lg:text-5xl xl:text-5xl font-black gradient"
        >
          مسیر صعود به قله‌های برنامه‌نویسی
        </h1>
        <p
          class="max-w-2xl  text-lg md:text-xl font-bold leading-8"
        >
          هر جای مسیرِ برنامه‌نویسی که باشی، با هم‌راهی استادهای باتجربهٔ کلاسبن می‌تونی بدون محدودیت به قله‌های بالاتر صعود کنی. ما همیشه هواتو داریم.
        </p>
        <div
          class="mt-5 flex gap-4"
        >
          <button
            class="btn btn-primary btn-lg "
            type="button"
          >
            دوره‌های ری‌اکت و نکست
          </button>
          <button
            class="btn btn-neutral btn-lg "
            type="button"
          >
            مشاوره برنامه‌نویسی
          </button>
        </div>
        <img
          alt=""
          class="grayscale mt-4 opacity-70 m-auto xl:m-0"
          data-nimg="1"
          decoding="async"
          height="39"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75"
          srcset="/_next/image?url=%2Fimages%2Fframeworks.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fframeworks.png&w=828&q=75 2x"
          style="color: transparent;"
          width="412"
        />
      </div>
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="521"
        loading="lazy"
        src="/images/programmer-landing.svg"
        style="color: transparent;"
        width="702"
      />
    </div>
  </section>
</div>
`;
